nexong/api/Event/tests.py

```diff
@@ -370,3 +370,215 @@ def test_delete_lesson_event_by_admin(self):
+
+        self.voluntario = Volunteer.objects.create(
+            academic_formation="Test volunteer",
+            motivation="Test volunteer",
+            status="PENDIENTE",
+            address="Test address3",
+            postal_code=12350,
+            birthdate="1956-07-06",
+            start_date="1956-07-06",
+            end_date="1956-07-07",
+        )
+        self.voluntario2 = Volunteer.objects.create(
+            academic_formation="Test volunteer7",
+            motivation="Test volunteer7",
+            status="ACEPTADO",
+            address="Test address21",
+            postal_code=12349,
+            birthdate="1956-07-05",
+            start_date="1956-07-05",
+            end_date="1956-07-05",
+        )
+
+        self.event2 = Event.objects.create(
+            name="Feria",
+            description="Se necesitan 2 camareros para atenderlas",
+            place="Cacharritos1",
+            max_volunteers=2,
+            max_attendees=2,
+            price=5,
+            start_date="2024-06-13 06:00-00:00",
+            end_date="2024-06-13 11:00-00:00",
+        )
+        self.event2.attendees.add(self.student, self.student2)
+        self.event2.volunteers.add(self.voluntario, self.voluntario2)
+        self.lessonevent2 = LessonEvent.objects.create(
+            name="Vienen los del Ríopol",
+            description="Se necesitan educadoresad",
+            place="Patio Centrales",
+            max_volunteers=2,
+            price=25.0,
+            lesson=self.lesson,
+            start_date="2025-06-13T05:00:00Z",
+            end_date="2025-06-13T16:00:00Z",
+        )
+
+    def test_obtain_event_by_volunteer(self):
+        response = self.client.get(
+            f"/api/event/{self.event2.id}/",
+            HTTP_AUTHORIZATION=f"Token {self.token.key}",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_create_event_by_volunteer_error(self):
+        attendees_ids1 = [self.student.id, self.student2.id]
+        volunteers_ids1 = [
+            self.voluntario.id,
+            self.voluntario2.id,
+        ]  # Asegurarse de que se esté pasando el ID del voluntario
+        response = self.client.post(
+            "/api/event/",
+            data={
+                "name": "Viene la familia Peruana",
+                "description": "Se necesitan gente",
+                "place": "La cocina",
+                "max_volunteers": 2,
+                "max_attendees": 2,
+                "price": 5,
+                "start_date": "2025-06-12T06:00:00Z",
+                "end_date": "2025-06-12T11:00:00Z",
+                "attendees": attendees_ids1,
+                "volunteers": volunteers_ids1,
+            },
+            HTTP_AUTHORIZATION=f"Token {self.token.key}",
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_update_event_by_volunteer_error(self):
```

Comment:
Both this function and test_update_lesson_event_by_volunteer are actually what a Volunteer should be allowed to do. Because as I said, the volunteer should only be allowed to PUT the volunteers field of an Event/LessonEvent, which is how they sign up for events. But they are not allowed to update other fields, that is why you get the 403 error.